libraries/AP_OpticalFlow/OpticalFlow.cpp

```diff
@@ -41,32 +41,35 @@ const AP_Param::GroupInfo OpticalFlow::var_info[] = {
 };
 
 // default constructor
-OpticalFlow::OpticalFlow(AP_AHRS_NavEKF& ahrs) :
-#if CONFIG_HAL_BOARD == HAL_BOARD_PX4
-    backend(new AP_OpticalFlow_PX4(*this)),
-#elif CONFIG_HAL_BOARD == HAL_BOARD_SITL
-    backend(new AP_OpticalFlow_HIL(*this)),
-#elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BEBOP ||\
-      CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_MINLURE ||\
-      CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
-    backend(new AP_OpticalFlow_Onboard(*this, ahrs)),
-#elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX
-    backend(new AP_OpticalFlow_Linux(*this, hal.i2c_mgr->get_device(HAL_OPTFLOW_PX4FLOW_I2C_BUS, HAL_OPTFLOW_PX4FLOW_I2C_ADDRESS))),
-#else
-    backend(NULL),
-#endif
-    _last_update_ms(0)
+OpticalFlow::OpticalFlow(AP_AHRS_NavEKF &ahrs)
+    : _last_update_ms(0)
 {
     AP_Param::setup_object_defaults(this, var_info);
 
     memset(&_state, 0, sizeof(_state));
 
     // healthy flag will be overwritten on update
     _flags.healthy = false;
-};
+
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BEBOP ||\
+    CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_MINLURE ||\
+    CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
+    backend = new AP_OpticalFlow_Onboard(*this, ahrs);
+#endif
```

Comment:
because ahrs is only available in the constructor and I didn't want to store a reference to it in the class.

I think in the end we will need to change this not to pass a reference to ahrs, so I didn't bother to be consistent here.